Classes/Service/ContentService.php

```diff
@@ -345,24 +345,38 @@ protected function initializeRecordByNewAndOldAndLanguageUids(
                 // look for the translated version of the parent record indicated
                 // in this new, translated record. Below, we adjust the parent UID
                 // so it has the UID of the translated parent if one exists.
-                $translatedParents = (array) $this->workspacesAwareRecordService->get(
-                    'tt_content',
-                    'uid,sys_language_uid',
-                    "t3_origuid = '" . $oldRecord['tx_flux_parent'] . "'" . BackendUtility::deleteClause('tt_content')
-                );
-                foreach ($translatedParents as $translatedParent) {
-                    if ($translatedParent['sys_language_uid'] == $newLanguageUid) {
-                        // set $translatedParent to the right language ($newLanguageUid):
-                        break;
+                unset($copiedParent);
+                if (0 < (integer) $oldRecord['tx_flux_parent']) {
+                    // Search for a container element which has been copied from
+                    // the container referred to by the original element
+                    // (oldRecord) and exists in the target language.
+                    // Note that this cannot guarantee the correct container
+                    // to be found and used (using copies for translations
+                    // leads to ambiguous results by principle).
+                    // This is the old method which used to implement a fix for
+                    // #816 and #796.
+                    $copiedParents = (array) $this->workspacesAwareRecordService->get(
+                        'tt_content',
+                        'uid,sys_language_uid',
+                        "t3_origuid = '" . (integer) $oldRecord['tx_flux_parent'] . "' AND " .
+                            "pid = '" . (integer) $row['pid'] . "' AND " .
+                            "CType = 'fluidcontent_content' AND " .
```

Comment:
It's not super obvious but you could in theory turn a `news` plugin into a content container. Use case example could be wanting to put unique content elements which show as default if no news exist (basically: anything you an imagine that can use a relation to a content record and could benefit from editing that relation as a nested content area). Such a conversion can even be done without touching a single line of code or template from the `news` extension (short version: register Provider for CType/list_type, define Grid in form). It is even possible to do this for non-content records; allowing them to contain a Grid of related content - however, such non-content records cannot be edited in the page module by default. Flux was built with the intention that *any* record can become a container (but **child** content is always from `tt_content` since this is the type of relation Flux was designed for). Hopefully that explains why there are no CType restrictions in Flux itself ;)

> Is there any better (and still easy) way to do a basic check if the chosen records are capable of being used as containers or should that extra check simply be removed?

I think we would need to remove it. There *is* a way but it isn't easy - you have to use the Flux API to filter records. The way that would be done is you would detect a Provider for each `$copiedParents` record and ask that Provider (if one can be resolved) for a Grid, then check if that Grid contains at least one row and one column. Looking at it without having tested, leaving out the CType clause should still work just as well.